### PR TITLE
Add tests for RateDefinition arithmetic

### PR DESCRIPTION
**Pull request by dannyroberts**
##### SUMMARY
I realized there was some doubt in my mind as to whether this new rate limiting logic was working the way I thought it was, so I figured I would go ahead and add tests for the mathier part of the whole thing since that is really easy to test.

The tests passed without any modifications to the production code, but I'm much more confident in what the code does now that I wrote these tests.

### CODE REVIEW

**Review by millerdev — APPROVED** [review]
Using randomness in tests means a test could fail intermittently and it may be difficult to tell how to reproduce that failure. I almost never use randomness in tests because of this. Would it be possible to come up with one or two common cases plus some edge cases and eliminate the randomness?

**Comment by dannyroberts** [comment]
I see what you're saying about randomness. I figured since I was testing something that was fairly straight forward arithmetic, in my mind it was like writing a test to show that x + x is always the same as 2 * x, but plugging in random float values for x—if it ever worked with 15 random choices for x, it's unlikely there's _any_ choice that doesn't work. (Of course if I were testing x / x = 1, my analogy breaks down because it'll work find until I randomly choose 0.)

I'm heading out for the day but I'll revisit this shortly, since I think the point you're raising makes sense

**Comment by dannyroberts** [comment]
I do think I will merge this in as is today so I can update tests as I make changes here, but I will keep it on my near term to revisit the randomness. I agree with the overall point and while it was helpful when initially putting together the tests, I could replace it with some well-chosen examples

**Comment by dannyroberts** [comment]
(build fail unrelated)

File: corehq/project_limits/tests/test_rate_definition.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        per_minute=0.07,
+        per_second=0.005,
+    )
+    floor_rate = RateDefinition(
+        per_day=50,
+        per_hour=30,
+        per_minute=10,
+        per_second=1,
+    )
+    for n_users, per_second, per_minute, per_hour, per_day, per_week in table:
```
Comment (millerdev) [inline corehq/project_limits/tests/test_rate_definition.py]: Could also use a generator test to make each row into a separate test.

Comment (dannyroberts) [reply corehq/project_limits/tests/test_rate_definition.py]: Good catch. Tacking on here: https://github.com/dimagi/commcare-hq/pull/25395/commits/3b1666f1f92d762b688501fea4d3bd348c3c644c

**Pull request merged**